apidoc/Titanium/UI/ButtonBar.yml

```diff
@@ -42,6 +42,11 @@ events:
         type: Number
 properties:
   - name: index
+    deprecated:
+        since: "8.0.0"
+        notes: |
+            This property has been removed in Titanium SDK 8.0.0 as this property has no effect.
+        removed: "8.0.0"
```

Comment:
The "index" property isn't actually removed. It's just ignored and logs a warning now.

@janvennemann, for your info, the "index" property was added to `Ti.UI.ButtonBar` by mistake because it shares most of the same code as `Ti.UI.OptionBar`. The "index" property only makes sense for `OptionBar` since only 1 button in the segmented control can be selected at a time (like a radio button group).

A `ButtonBar` is merely a horizontal row of buttons. The "index" property is meaningless for this control. It doesn't do anything useful. So, this is more like a bug fix. I suppose we can keep the property and have it do nothing instead?

Reply:
We should still keep our docs and code in sync. And since it was not removed then marking it as removed in 8.0.0 is wrong and it should only be marked as deprecated.

@vijaysingh-axway cc

Reply:
@janvennemann @jquick-axway 
1. If someone is using 'index' property of Ti.UI.ButtonBar, they would be getting no effect or wrong effect. So this property is useless. 
2.  As for as deprecation and removal of a property in same version is concerned, neither we are going to give any substitute for that property nor it is giving any desired effect. So I think we can remove that in same version. 

Or
Should I mark message in doc and code as deprecate only ?

Reply:
I'd say update the message to say it will be removed in 9.0.0 and then we remove it there.